src/kit/Button.tsx

```diff
@@ -13,18 +13,18 @@ interface ButtonProps {
   children?: ReactNode;
   danger?: boolean;
   disabled?: boolean;
-  form?: string;
-  hideChildren?: boolean;
+  form?: string | undefined;
+  hideChildren?: boolean | undefined;
   htmlType?: 'button' | 'submit' | 'reset';
   icon?: ReactNode;
-  column?: boolean;
-  loading?: boolean | { delay?: number };
+  column?: boolean | undefined;
+  loading?: boolean | { delay?: number } | undefined;
   onClick?: (event: MouseEvent) => void;
-  ref?: React.Ref<HTMLElement>;
-  selected?: boolean;
+  ref?: React.Ref<HTMLElement> | undefined;
+  selected?: boolean | undefined;
   size?: 'large' | 'middle' | 'small';
   type?: 'primary' | 'text' | 'default' | 'dashed';
-  tooltip?: string;
+  tooltip?: string | undefined;
```

Comment:
is there a guideline for when to add `| undefined` that's being applied here? not sure I understand why, for example, `column` allows `undefined` but `disabled` does not.

Reply:
it is because antd does not take `undefined` for some props. 
For example, if a props is like `foo? string`, we cannot pass `undefined` with `exactOptionalPropertyTypes` option. The prop must be `foo?: string | undefined` if we want to pass `undefined`. 
I pretty much followed the antd props, so `disabled` doesn't let us pass `undefined`. If we wanna accept `undefined` for `disable`, we can do it and add default value in case the value is `undefined`